### PR TITLE
Add node source tests without InfrastructureManager modification

### PR DESCRIPTION
**Pull request by jrochas**
- this commit is the same as 35bfa57e4bbf44b5c01391926b366670c257b4c5, but does not remove a method needed by a dependency (nodesource-addons).

### CODE REVIEW

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 4 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [InfrastructureManager.java#L302](https://github.com/ow2-proactive/scheduling/blob/ec9c0f06a10a410a0ab9355c3c833311890b13fd/rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/InfrastructureManager.java#L302): Define a constant instead of duplicating this literal "DeployingNode " 3 times. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)
1. ![MINOR][MINOR] [InfrastructureManager.java#L1070](https://github.com/ow2-proactive/scheduling/blob/ec9c0f06a10a410a0ab9355c3c833311890b13fd/rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/InfrastructureManager.java#L1070): Reorder the modifiers to comply with the Java Language Specification. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AModifiersOrderCheck)
1. ![MINOR][MINOR] [InfrastructureManager.java#L1071](https://github.com/ow2-proactive/scheduling/blob/ec9c0f06a10a410a0ab9355c3c833311890b13fd/rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/InfrastructureManager.java#L1071): Rename this field "DEFAULT" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3008)
1. ![MINOR][MINOR] [InfrastructureManager.java#L1084](https://github.com/ow2-proactive/scheduling/blob/ec9c0f06a10a410a0ab9355c3c833311890b13fd/rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/InfrastructureManager.java#L1084): Use a logger to log this exception. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1148)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [InfrastructureManager.java#L1070](https://github.com/ow2-proactive/scheduling/blob/7096fb75e32785b5e4e13f0ef191efa8ca832e8e/rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/InfrastructureManager.java#L1070): Reorder the modifiers to comply with the Java Language Specification. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AModifiersOrderCheck)
1. ![MINOR][MINOR] [InfrastructureManager.java#L1071](https://github.com/ow2-proactive/scheduling/blob/7096fb75e32785b5e4e13f0ef191efa8ca832e8e/rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/InfrastructureManager.java#L1071): Rename this field "DEFAULT" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3008)
1. ![MINOR][MINOR] [InfrastructureManager.java#L1084](https://github.com/ow2-proactive/scheduling/blob/7096fb75e32785b5e4e13f0ef191efa8ca832e8e/rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/InfrastructureManager.java#L1084): Use a logger to log this exception. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1148)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/InfrastructureManager.java [hunk 1]
```diff
@@ -1141,192 +1144,120 @@ protected abstract RMDeployingNode newRMDeployingNode(String name, NodeSource ns
...
     private void setUsingDeployingNodesWithLockAndPersist() {
-        setPersistedInfraVariable(new PersistedInfraVariablesHandler<Void>() {
-            @Override
-            public Void handle() {
-                persistedInfraVariables.put(USING_DEPLOYING_NODES_KEY, true);
-                return null;
-            }
+        setPersistedInfraVariable((PersistedInfraVariablesHandler<Void>) () -> {
+            this.persistedInfraVariables.put(USING_DEPLOYING_NODES_KEY, true);
+            return null;
```
Comment (MaelAudren) [inline rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/InfrastructureManager.java]: Can you remove "return null" ?

File: rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/InfrastructureManager.java [hunk 2]
```diff
@@ -1141,192 +1144,120 @@ protected abstract RMDeployingNode newRMDeployingNode(String name, NodeSource ns
...
     protected void setRmUrl(final String rmUrl) {
-        setPersistedInfraVariable(new PersistedInfraVariablesHandler<Void>() {
-            @Override
-            public Void handle() {
-                persistedInfraVariables.put(RM_URL_KEY, rmUrl);
-                return null;
-            }
+        setPersistedInfraVariable((PersistedInfraVariablesHandler<Void>) () -> {
+            this.persistedInfraVariables.put(RM_URL_KEY, rmUrl);
+            return null;
```
Comment (MaelAudren) [inline rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/InfrastructureManager.java]: Can you remove "return null" ?

File: rm/rm-server/src/test/java/functionaltests/nodesource/TestLocalInfrastructureCronPolicy.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    @Before
+    public void setup() throws Exception {
+        this.resourceManager = this.rmHelper.getResourceManager();
+        this.monitorsHandler = this.rmHelper.getMonitorsHandler();
+    }
+
+    @Test
+    public void action() throws Exception {
+
+        log("Create an remove an empty local node source with cron policy");
```
Comment (MaelAudren) [inline rm/rm-server/src/test/java/functionaltests/nodesource/TestLocalInfrastructureCronPolicy.java]: "Create an remove an empty local node" doesn't make sense

File: rm/rm-server/src/test/java/functionaltests/nodesource/TestLocalInfrastructureTimeSlotPolicy.java [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    @Before
+    public void setup() throws Exception {
+        this.resourceManager = this.rmHelper.getResourceManager();
+        this.monitorsHandler = this.rmHelper.getMonitorsHandler();
+    }
+
+    @Test
+    public void action() throws Exception {
+
+        log("Create an remove an empty local node source with time slot policy");
```
Comment (MaelAudren) [inline rm/rm-server/src/test/java/functionaltests/nodesource/TestLocalInfrastructureTimeSlotPolicy.java]: "Create an remove an empty local node" doesn't make sense

File: rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/InfrastructureManager.java (hunk 1 above)
Comment (jrochas) [reply rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/InfrastructureManager.java]: Yes I can clean this code.

File: rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/InfrastructureManager.java (hunk 2 above)
Comment (jrochas) [reply rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/InfrastructureManager.java]: Yes I can clean this code.

File: rm/rm-server/src/test/java/functionaltests/nodesource/TestLocalInfrastructureCronPolicy.java (hunk 3 above)
Comment (jrochas) [reply rm/rm-server/src/test/java/functionaltests/nodesource/TestLocalInfrastructureCronPolicy.java]: Corrected.

File: rm/rm-server/src/test/java/functionaltests/nodesource/TestLocalInfrastructureTimeSlotPolicy.java (hunk 4 above)
Comment (jrochas) [reply rm/rm-server/src/test/java/functionaltests/nodesource/TestLocalInfrastructureTimeSlotPolicy.java]: Corrected.

File: rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/InfrastructureManager.java [hunk 5]
```diff
@@ -1173,7 +1173,7 @@ private boolean containsDeployingNodeWithLock(final String nodeUrl) {
     }
 
     private void clearDeployingNodesWithLockAndPersist() {
-        setPersistedInfraVariable((PersistedInfraVariablesHandler<Void>) () -> {
+        setPersistedInfraVariable(() -> {
             this.deployingNodesMap.clear();
             getPersistedDeployingNodesUrl().clear();
             return null;
```
Comment (MaelAudren) [inline rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/InfrastructureManager.java]: remove "return null"

File: rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/InfrastructureManager.java [hunk 6]
```diff
@@ -1141,192 +1144,108 @@ protected abstract RMDeployingNode newRMDeployingNode(String name, NodeSource ns
...
-            @Override
-            public Void handle() {
-                deployingNodesMap.clear();
-                getPersistedDeployingNodesUrl().clear();
-                return null;
-            }
+        setPersistedInfraVariable(() -> {
+            this.deployingNodesMap.clear();
+            getPersistedDeployingNodesUrl().clear();
+            return null;
```
Comment (MaelAudren) [inline rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/InfrastructureManager.java]: Remove return null

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [InfrastructureManager.java#L1070](https://github.com/ow2-proactive/scheduling/blob/e33e6559ddc21594e98c74e813fdbe3d93be1fd6/rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/InfrastructureManager.java#L1070): Reorder the modifiers to comply with the Java Language Specification. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AModifiersOrderCheck)
1. ![MINOR][MINOR] [InfrastructureManager.java#L1071](https://github.com/ow2-proactive/scheduling/blob/e33e6559ddc21594e98c74e813fdbe3d93be1fd6/rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/InfrastructureManager.java#L1071): Rename this field "DEFAULT" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3008)
1. ![MINOR][MINOR] [InfrastructureManager.java#L1084](https://github.com/ow2-proactive/scheduling/blob/e33e6559ddc21594e98c74e813fdbe3d93be1fd6/rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/InfrastructureManager.java#L1084): Use a logger to log this exception. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1148)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/InfrastructureManager.java (hunk 6 above)
Comment (jrochas) [reply rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/InfrastructureManager.java]: This is unfortunately not possible as it won't compile.

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [InfrastructureManager.java#L1070](https://github.com/ow2-proactive/scheduling/blob/721e41a2d3346df23a3446d82e6ea86b620fa079/rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/InfrastructureManager.java#L1070): Reorder the modifiers to comply with the Java Language Specification. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AModifiersOrderCheck)
1. ![MINOR][MINOR] [InfrastructureManager.java#L1071](https://github.com/ow2-proactive/scheduling/blob/721e41a2d3346df23a3446d82e6ea86b620fa079/rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/InfrastructureManager.java#L1071): Rename this field "DEFAULT" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3008)
1. ![MINOR][MINOR] [InfrastructureManager.java#L1084](https://github.com/ow2-proactive/scheduling/blob/721e41a2d3346df23a3446d82e6ea86b620fa079/rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/InfrastructureManager.java#L1084): Use a logger to log this exception. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1148)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [InfrastructureManager.java#L1070](https://github.com/ow2-proactive/scheduling/blob/ffc52c9d622aa4d5b965b024af5ef4fc340e4f84/rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/InfrastructureManager.java#L1070): Reorder the modifiers to comply with the Java Language Specification. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AModifiersOrderCheck)
1. ![MINOR][MINOR] [InfrastructureManager.java#L1071](https://github.com/ow2-proactive/scheduling/blob/ffc52c9d622aa4d5b965b024af5ef4fc340e4f84/rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/InfrastructureManager.java#L1071): Rename this field "DEFAULT" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3008)
1. ![MINOR][MINOR] [InfrastructureManager.java#L1084](https://github.com/ow2-proactive/scheduling/blob/ffc52c9d622aa4d5b965b024af5ef4fc340e4f84/rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/InfrastructureManager.java#L1084): Use a logger to log this exception. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1148)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**